### ISSUE TITLE
Infinite Indexing Loop in Intellij 2023.3

### ISSUE THREAD

**Issue by lbenedetto** [fan-tom/intellij-codeowners#209]
Plugin version: 0.6.0
IntelliJ Version: 2023.3
```
023-12-01 15:36:33,817 [2123663]   INFO - #c.i.u.i.d.AppIndexingDependenciesService - Invalidating all indexing flags. Reason: Rebuild requested for index CodeownersFilesIndex
2023-12-01 15:36:33,821 [2123667]   INFO - #c.i.u.i.FileBasedIndexImpl - Rebuild requested for index CodeownersFilesIndex
com.intellij.util.indexing.StorageException: java.lang.NoClassDefFoundError: org/apache/commons/lang/builder/HashCodeBuilder
	at com.intellij.util.indexing.impl.MapReduceIndex.updateWithMap(MapReduceIndex.java:372)
	at com.intellij.util.indexing.storage.MapReduceIndexBase.updateWithMap(MapReduceIndexBase.java:102)
	at com.intellij.util.indexing.impl.MapReduceIndex$IndexUpdateComputable.compute(MapReduceIndex.java:401)
	at com.intellij.util.indexing.impl.MapReduceIndex$IndexUpdateComputable.compute(MapReduceIndex.java:383)
	at com.intellij.openapi.util.Computable.get(Computable.java:16)
	at com.intellij.indexing.composite.CompositeInvertedIndexBase.lambda$mapInputAndPrepareUpdate$3(CompositeInvertedIndexBase.java:77)
	at com.intellij.util.indexing.IndexInfrastructureExtensionUpdateComputation.compute(IndexInfrastructureExtensionUpdateComputation.java:27)
	at com.intellij.util.indexing.IndexInfrastructureExtensionUpdateComputation.compute(IndexInfrastructureExtensionUpdateComputation.java:11)
	at com.intellij.openapi.util.Computable.get(Computable.java:16)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$runUpdateForPersistentData$24(FileBasedIndexImpl.java:1638)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$3(CoreProgressManager.java:236)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:53)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:236)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$runUpdateForPersistentData$25(FileBasedIndexImpl.java:1638)
	at com.intellij.util.indexing.StorageBufferingHandler.runUpdate(StorageBufferingHandler.java:22)
	at com.intellij.util.indexing.FileBasedIndexImpl.runUpdateForPersistentData(FileBasedIndexImpl.java:1637)
	at com.intellij.util.indexing.SingleIndexValueApplier.doApply(SingleIndexValueApplier.java:62)
	at com.intellij.util.indexing.SingleIndexValueApplier.apply(SingleIndexValueApplier.java:50)
	at com.intellij.util.indexing.FileIndexesValuesApplier.applyModifications(FileIndexesValuesApplier.java:199)
	at com.intellij.util.indexing.FileIndexesValuesApplier.lambda$apply$0(FileIndexesValuesApplier.java:171)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$2(CoreProgressManager.java:228)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$3(CoreProgressManager.java:236)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:57)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:236)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:227)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.lambda$scheduleIndexWriting$7(IndexUpdateRunner.java:781)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:244)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:30)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:222)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:218)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:210)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/lang/builder/HashCodeBuilder
	at com.github.fantom.codeowners.indexing.CodeownersEntryOccurrence.hashCode(CodeownersEntryOccurrence.kt:85)
	at java.base/java.util.HashMap.hash(HashMap.java:338)
	at java.base/java.util.HashMap.put(HashMap.java:610)
	at com.intellij.util.indexing.InputMapExternalizer.save(InputMapExternalizer.java:59)
	at com.intellij.util.indexing.InputMapExternalizer.save(InputMapExternalizer.java:24)
	at com.intellij.util.indexing.impl.forward.AbstractForwardIndexAccessor.serializeToByteSeq(AbstractForwardIndexAccessor.java:74)
	at com.intellij.util.indexing.impl.forward.AbstractForwardIndexAccessor.serializeValueToByteSeq(AbstractForwardIndexAccessor.java:63)
	at com.intellij.util.indexing.impl.forward.AbstractForwardIndexAccessor.serializeIndexedData(AbstractForwardIndexAccessor.java:44)
	at com.intellij.util.indexing.impl.forward.AbstractForwardIndexAccessor.serializeIndexedData(AbstractForwardIndexAccessor.java:39)
	at com.intellij.util.indexing.impl.MapReduceIndex.updateForwardIndex(MapReduceIndex.java:288)
	at com.intellij.util.indexing.impl.storage.TransientFileContentIndex.updateForwardIndex(TransientFileContentIndex.java:78)
	at com.intellij.util.indexing.impl.MapReduceIndex.lambda$prepareUpdate$4(MapReduceIndex.java:273)
	at com.intellij.util.indexing.impl.UpdateData.updateForwardIndex(UpdateData.java:63)
	at com.intellij.util.indexing.impl.MapReduceIndex.updateWithMap(MapReduceIndex.java:365)
	... 40 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.builder.HashCodeBuilder PluginClassLoader(plugin=PluginDescriptor(name=CODEOWNERS, id=fantom.codeowners, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2023.3/CODEOWNERS, version=0.6.0, package=null, isBundled=false), packagePrefix=null, state=active)
	... 54 more
```

**Comment by matan129** [comment]
Additional context - https://youtrack.jetbrains.com/issue/IDEA-309605

It seems that when this plugin is enabled in 2023.3, indexing never stops

**Comment by YannCebron** [comment]
This is a regression in IJ platform that will be fixed (missing org/apache/commons/lang/builder/HashCodeBuilder class) in upcoming 2023.3 bugfix releases.

As workaround, bundle commons-lang library explicitly with the plugin.

Sorry for the inconvenience.

**Comment by fan-tom** [comment]
> This is a regression in IJ platform that will be fixed (missing org/apache/commons/lang/builder/HashCodeBuilder class) in upcoming 2023.3 bugfix releases.

From where I can know about such things? I thought they simply got rid of this dependency (intentionally), so I rewrote the code to not use it in 1f65d248

**Comment by fan-tom** [comment]
Fixed in v0.7.0

**Comment by YannCebron** [comment]
As I wrote above, it was a regression and not intentional complete removal of this library.https://plugins.jetbrains.com/docs/intellij/api-changes-list-2023.html#b29cd76b_40
